mt6625l: some fixes for R2 internal WiFi driver

lexa2 opened the pull request:
Signed-off-by: Alexey Loukianov <lx2@lexa2.ru>

lexa2 committed:
mt6625l: use proper decorations for module init/exit

Signed-off-by: Alexey Loukianov <lx2@lexa2.ru>

lexa2 committed:
mt6625l: fix/impove driver debugging features

Signed-off-by: Alexey Loukianov <lx2@lexa2.ru>

lexa2 committed:
mt6625l: create phy only when device is added

Original-Author: Nikolay Amiantov <ab@fmap.me>
Signed-off-by: Alexey Loukianov <lx2@lexa2.ru>

lexa2 committed:
mt6625l: fix building as a module

Signed-off-by: Alexey Loukianov <lx2@lexa2.ru>

lexa2 committed:
mt6625l: restore use of wakeup source in irq handler

Signed-off-by: Alexey Loukianov <lx2@lexa2.ru>

frank-w commented on drivers/misc/Makefile:
```diff
@@ -57,4 +57,4 @@ obj-y				+= cardreader/
 obj-$(CONFIG_HABANA_AI)		+= habanalabs/
 obj-$(CONFIG_XILINX_SDFEC)	+= xilinx_sdfec.o
-obj-$(CONFIG_MTK_COMBO)	+= mediatek/
+obj-y				+= mediatek/
```
This looks wrong to me...it always includes driver if misc-folder used

lexa2 replied:
MTK_COMBO is mutually independent with MTK_BTIF. It is normal situation to have btif built-in while other parts are disabled or built as a module. Relying here solely on MTK_COMBO results in kbuild either skipping btif or building it but failing to include its objects and export its symbols in kernel. End result is build failing at modules MODPOST step with unresolved symbols in built mtk modules due to missing btif.

frank-w replied:
why not set MTK_BTIF via Kconfig if MTK_COMBO is set (select-statement)

if somebody does not use wifi-driver it always gets some parts build into (if misc-folder is included by upper makefile)

lexa2 replied:
Simple `select  MTK_BTIF` in MTK_COMBO don't work for the case when we have MTK_COMBO=m and MTK_BTIF=y. Solution is to use `obj-$(CONFIG_MTK_BTIF)` in misc toplevel makefile instead of CONFIG_MTK_COMBO and add `select MTK_BTIF` to MTK_COMBO Kconfig.

frank-w reviewed (commented):
Looks good so far...have you tested function as builtin and module?

lexa2 commented:
Re normalizing tags - sure, will do.

lexa2 commented:
Normalization: done.

lexa2 commented:
Frank, I had updated this PR to match the current state of your 5.4-wifi branch. Please check if I had addressed all points from your comments and feel free to merge this one if everything seems to be more or less okay. Thanks.

frank-w commented:
Hi
thanks for your work,have you found the memory issue? Had not time yet to test by myself :(

lexa2 commented:
Hadn't had any luck with tracing potention memory corruption issue when built as a module - last time I was testing wifi as a module board had been working fine with 6 days uptime without any issues. So it seems to be hard to track intermittent issue.

lexa2 committed:
mt6625l: fix out-of-tree build (i.e. when KBUILD_OUTPUT env is set)

Signed-off-by: Alexey Loukianov <lx2@lexa2.ru>

frank-w commented:
already merged to 5.4-main, thank you ;)
